File: C++/LocalizationTest.cpp
#include "Localization.h"
#include <iostream>
#include <cstdlib>
#include <string>
using namespace std;

int main() {
	cout << "Tests for SimpleLion - C++ module (C) 2013 SimpleLion Development Team" << endl;
	cout << "Localization test (main test) v. 0.1" << endl;
	std::string wd = getenv("PWD");
	std::string parsed = "";
	std::string upperwd = "";
	for(int i = 0; i < wd.length(); i++) {
		if(wd[i] == '/') {
			upperwd += parsed;
			upperwd += wd[i];
			parsed = "";
		} else {
			parsed += wd[i];
		}
	}
	wd = upperwd + "locale";
	try {
		SimpleLion::Localization loc(wd);
		cout << "Localization object up and working. What next?" << endl << endl;
		int action = 0;
		while(action != 5) {
			cout << "1) Load a localization file" << endl;
			cout << "2) Query for a translation" << endl;
			cout << "3) Display locale list" << endl;
			cout << "4) Convert Localization back to SLLF and display" << endl;
			cout << "5) Exit test" << endl;
			cout << "Select action: ";
			cin >> action;
			string locale;
			string squery;
			string* locales;
			switch(action) {
			case 1:
				cout << "Enter locale code (e.g. en, pl etc.): ";
				cin >> locale;
				loc.setLocale(locale);
				cout << "Locale successfully loaded!" << endl;
				break;
			case 2:
				cout << "Enter query string: ";
				cin >> squery;
				cout << endl << "Query returned: " << loc.query(squery) << endl;
				break;
			case 3:
				locales = loc.localeList();
				cout << "Available locales:" << endl;
				for(int i = 0; i < loc.localeListSize(); i++) {
					cout << locales[i] << endl;
				}
				break;
			case 4:
				cout << loc.toSLLF();
				break;
			case 5:
				break;
			}
		}
	} catch(SimpleLion::FilesystemException &exc) {
		cout << "FilesystemException: " << exc.what() << endl;
		return 3;
	} catch(SimpleLion::FileException &exc) {
		cout << "FileException: " << exc.what() << endl;
		return 2;
	} catch(exception &exc) {
		cout << "Exception: " << exc.what() << endl;
		return EXIT_FAILURE;
	}
	cout << "END OF TEST" << endl;
	return EXIT_SUCCESS;
}

File: PHP/localization.test.php
<?php
include("localization.inc.php");
print "SimpleLion PHP module tests - (C) 2013 SimpleLion Development Team\n";
print "Localization (main class) test v.0.1\n";
try {
	chdir('..');
	chdir('locale');
	$dir = getcwd();
	print "Got directory: ".$dir."\n";
	print "Constructing Localization object...\n";
	$loc = new SimpleLion\Localization($dir);
	print "Localization object up and working!\n";
	$locales = $loc->localeList();
	print "Loaded locale list:\n";
	print_r($locales);
	for($i = 0; $i < count($locales); $i++) {
		print "Loading locale: ".$locales[$i]."\n";
		$loc->setLocale($locales[$i]);
		print "test: ".$loc->query("test")."\n";
		print "test2: ".$loc->query("test2")."\n";
		print "test3: ".$loc->query("test3")."\n";
		print "test4: ".$loc->query("test4")."\n";
		print "testcat.test: ".$loc->query("testcat.test")."\n";
		print "testcat.test2: ".$loc->query("testcat.test2")."\n";
		print "testcat.test3: ".$loc->query("testcat.test3")."\n";
		print "testcat.test4: ".$loc->query("testcat.test4")."\n";
		print "testcat.acat.test: ".$loc->query("testcat.acat.test")."\n";
		print "testcat.acat.test2: ".$loc->query("testcat.acat.test2")."\n";
		print "testcat: ".$loc->query("testcat")."\n";
		print "testcat.acat: ".$loc->query("testcat.acat")."\n";
		print "\n";
		print "This locale as SLLF:\n";
		print $loc->toSLLF();
	}
} catch(SimpleLion\FileException $e) {
	print "Caught FileException: ".$e."\n";
} catch(SimpleLion\FilesystemException $e) {
	print "Caught FilesystemException: ".$e."\n";
} catch(Exception $e) {
	print "Caught external Exception: ".$e->getMessage()."\n";
}
print "END OF TEST\n";
?>
